### DISCUSSION TITLE
Mac?

### DISCUSSION THREAD

**Discussion by Barugon** [cross-rs/cross#784] [Q&A] [answered]
How about some packages to cross compile to Mac? It could be done using [this](https://github.com/tpoechtrager/osxcross).

**Comment by Emilgardis** [comment, marked as answer]
We definitely want this.

Related:

https://github.com/cross-rs/cross/issues/223
https://github.com/cross-rs/cross/issues/508

**Reply by Emilgardis** [reply]
https://github.com/cross-rs/cross/discussions/784#discussioncomment-3018051

Implemented to the full extent possible

**Comment by Alexhuszagh** [comment]
I believe we can provide Dockerfiles to build the images, provide clear examples how to build them, but I don't believe we can distribute the generated images since they will contain the macOS [SDK](https://www.apple.com/legal/sla/docs/xcode.pdf).

Some choice quotes from there:

> Install a reasonable number of copies of the Apple Software on Apple-branded computers that
are owned or controlled by You to be used internally by You or Your Authorized Developers only as
follows:

> macOS SDK” means the Apple-proprietary Software Development Kits (SDKs) provided hereunder,
including but not limited to header files, APIs, libraries, simulators, and software (source code and object
code) labeled as part of the macOS SDK and included in the Xcode Developer Tools package for
purposes of targeting Apple-branded products running macOS

> This Agreement does not allow the Apple Software or Services to be made available over a
network where they could be run or used by multiple computers at the same time, unless otherwise
expressly permitted in writing by Apple. Further, unless otherwise expressly permitted by Apple in
writing, You agree not to rent, lease, lend, upload to or host on any website or server, sell, redistribute,
or sublicense the Apple Software and Apple Services, in whole or in part, or to enable others to do so.

So although the tools to install the SDK and a macOS build environment is legal, distributing images containing the macOS SDK is probably not legal. And I don't believe all of it falls under Apple's open source code. I don't think Apple is very strict in enforcing this, because you can find projects and images that **do** distribute their SDK quite readily and have for years.

**Reply by Emilgardis** [reply]
We can provide the cross-compiler though right? Without explicitly containing any Apple SDKs

**Reply by Alexhuszagh** [reply]
> 

I don't believe so, we need one specific one according to osxcross, and which I remember when attempting to create a Docker image of my own for my C++ version of cross. According to the [README](https://github.com/tpoechtrager/osxcross#how-does-it-work):

> For cross-compiling for macOS you need
>    the Clang/LLVM compiler
>    the cctools (lipo, otool, nm, ar, ...) and ld64
>    the macOS SDK.

**Reply by Emilgardis** [reply]
i see, what if we tell the user how to mount a legal copy?

**Reply by Alexhuszagh** [reply]
That we can do. And we can make it very easy for them to do so, all legally. All they would need to do is build the image, or have an image with everything besides the SDK to speed up the builds where the last step is installing the SDK which can be done by running the build script. As long as we are personally not hosting copies of the SDK, we should be fine.

**Reply by Alexhuszagh** [reply]
There are [plenty](https://github.com/phracker/MacOSX-SDKs) of [projects](https://github.com/phracker/MacOSX-SDKs) (the latter one is intentionally done under an alias) that do host these SDKs, so I doubt Apple will be litigious or enforce their copyright in this specific regard, but they could (and I'd rather not get on the bad side of Apple's lawyers).

**Reply by Barugon** [reply]
I can just use osxcross stand-alone.

**Reply by Alexhuszagh** [reply]
> Ah, that makes sense. I can just use osxcross stand-alone.

We definitely can make it easier though to use osxcross with this, however. And that should be done.

**Reply by Barugon** [reply]
I have to say that the idea of simply doing `cross build --release --target=x86_64-apple-darwin` is very appealing.

**Comment by Alexhuszagh** [comment]
This has been completed in #841, and a description on how to use the x86_64 and i686 Darwin targets can be found [here](https://github.com/cross-rs/cross/pull/480#issuecomment-1164982609).

**Reply by Barugon** [reply]
Awesome!
